loopin.go

```diff
@@ -969,6 +967,8 @@ func (s *loopInSwap) waitForSwapComplete(ctx context.Context,
 
 				invoiceFinalized = true
 				htlcKeyRevealed = s.tryPushHtlcKey(ctx)
+				s.cost.Server = s.AmountRequested -
```

Comment:
so before here, `s.cost.Server -= update.AmtPaid` would reach a negative amount, but then when processing the spend, we'd reach the positive server cost `s.cost.Server += htlcValue`? If that's the case then your change is a nice improvement.

Reply:
Yes exactly. I think the rationale was to keep it simple and account for changes as they happen, but it was very confusing.